components/function-controller/internal/webhook/resources/webhook_config.go

```diff
@@ -71,55 +74,102 @@ func ensureValidatingWebhookConfigFor(ctx context.Context, client ctlrclient.Cli
+				Namespace: config.ServiceNamespace,
+				Name:      config.ServiceName,
+				Path:      pointer.String("/defaulting/functions"),
+				Port:      pointer.Int32(443),
+			},
+		},
+		FailurePolicy:      &failurePolicy,
+		MatchPolicy:        &matchPolicy,
+		ReinvocationPolicy: &reinvocationPolicy,
+		Rules: []admissionregistrationv1.RuleWithOperations{
 			{
-				Name: DefaultingWebhookName,
-				AdmissionReviewVersions: []string{
-					"v1beta1",
-					"v1",
-				},
-				ClientConfig: admissionregistrationv1.WebhookClientConfig{
-					CABundle: config.CABundel,
-					Service: &admissionregistrationv1.ServiceReference{
-						Namespace: config.ServiceNamespace,
-						Name:      config.ServiceName,
-						Path:      pointer.String("/defaulting"),
-						Port:      pointer.Int32(443),
+				Rule: admissionregistrationv1.Rule{
+					APIGroups: []string{
+						serverlessAPIGroup,
 					},
-				},
-				FailurePolicy:      &failurePolicy,
-				MatchPolicy:        &matchPolicy,
-				ReinvocationPolicy: &reinvocationPolicy,
-				Rules: []admissionregistrationv1.RuleWithOperations{
-					{
-						Rule: admissionregistrationv1.Rule{
-							APIGroups: []string{
-								serverlessAPIGroup,
-							},
-							APIVersions: []string{
-								serverlessAPIVersion,
-							},
-							Resources: []string{"functions", "functions/status"},
-							Scope:     &scope,
-						},
-						Operations: []admissionregistrationv1.OperationType{
-							admissionregistrationv1.Create,
-							admissionregistrationv1.Update,
-						},
+					APIVersions: []string{
+						serverlessAPIVersion,
 					},
+					Resources: []string{"functions", "functions/status"},
+					Scope:     &scope,
+				},
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+					admissionregistrationv1.Update,
+				},
+			},
+		},
+		SideEffects:    &sideEffects,
+		TimeoutSeconds: pointer.Int32(30),
+	}
+}
+
+func getRegistryConfigSecretMutatingWebhook(config WebhookConfig) admissionregistrationv1.MutatingWebhook {
+	failurePolicy := admissionregistrationv1.Fail
+	matchPolicy := admissionregistrationv1.Exact
+	sideEffects := admissionregistrationv1.SideEffectClassNone
+
+	return admissionregistrationv1.MutatingWebhook{
+		Name: SecretMutationWebhookName,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			CABundle: config.CABundel,
+			Service: &admissionregistrationv1.ServiceReference{
+				Namespace: config.ServiceNamespace,
+				Name:      config.ServiceName,
+				Path:      pointer.String("/defaulting/registry-config-secrets"),
+				Port:      pointer.Int32(443),
+			},
+		},
+		FailurePolicy:           &failurePolicy,
+		MatchPolicy:             &matchPolicy,
+		TimeoutSeconds:          pointer.Int32(30),
+		SideEffects:             &sideEffects,
+		AdmissionReviewVersions: []string{"v1beta1", "v1"},
+		ObjectSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				RemoteRegistryLabelKey: "config",
```

Comment:
Would it be possible to define this in the function body? Technically is it's fine here, but it's not easy to find/read when placed here. And it's important.